### PR TITLE
Add multimap_from_entries function

### PR DESCRIPTION
**Pull request by wenleix**

### CODE REVIEW

**Comment by electrum** [comment]
Please add documentation

File: presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java [hunk 1]
```diff
@@ -860,6 +861,40 @@ public void testMapFromEntries()
         assertCachedInstanceHasBoundedRetainedSize("map_from_entries(ARRAY[('a', 1.0), ('b', 2.0), ('c', 3.0), ('d', 4.0), ('e', 5.0), ('f', 6.0)])");
     }
 
+    @Test
+    public void testMultimapFromEntries()
```
Comment (nezihyigitbasi) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: I ran a few simple tests and `select multimap_from_entries(ARRAY[null]);` fails with an IAE.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MultimapFromEntriesFunction.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import it.unimi.dsi.fastutil.ints.IntList;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+@ScalarFunction("multimap_from_entries")
+@Description("construct a map from an array of entries")
+public final class MultimapFromEntriesFunction
+{
+    private final PageBuilder pageBuilder;
+    private static final int INITIAL_LENGTH = 128;
```
[the remark this answers is not in the document]
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/MultimapFromEntriesFunction.java]: `INITIAL_ENTRY_COUNT`?

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MultimapFromEntriesFunction.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                entryIndicesList[keySet.positionOf(rowBlock, 0)].add(i);
+            }
+            else {
+                keySet.add(rowBlock, 0);
+                entryIndicesList[keySet.size() - 1].add(i);
+            }
+        }
+
+        BlockBuilder multimapBlockBuilder = pageBuilder.getBlockBuilder(0);
+        BlockBuilder singleMapWriter = multimapBlockBuilder.beginBlockEntry();
```
[the remark this answers is not in the document]
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/MultimapFromEntriesFunction.java]: @nezihyigitbasi : This seems to confused with the `multimapBlockBuilder` . The name also emphasize it's a `SingleMapWriter`

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MultimapFromEntriesFunction.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntList;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.google.common.base.Verify.verify;
+
+@ScalarFunction("multimap_from_entries")
+@Description("construct a map from an array of entries")
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MultimapFromEntriesFunction.java]: `map` -> `multimap`

**Pull request merged**